src/app/filter/filter.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { Filters, FiltersFormGroup } from './filter.model';
+import { FilterService } from './filter.service';
+
+@Component({
+  selector: 'app-filter',
+  templateUrl: './filter.component.html',
+  styleUrls: ['./filter.component.scss']
+})
+export class FilterComponent {
+  filterForm: FormGroup<FiltersFormGroup>;
+
+  constructor(private readonly filterService: FilterService) {
+    const { currentPage, entriesPerPage, nameFilter }: Filters = this.filterService.values$.value;
+
+    this.filterForm = new FormGroup<FiltersFormGroup>({
+      currentPage: new FormControl(currentPage),
+      entriesPerPage: new FormControl(entriesPerPage),
+      nameFilter: new FormControl(nameFilter)
+    });
```

Comment:
what about put this outside the constructor?
IIRC you could put anyways `filterForm: FormGroup<FiltersFormGroup>;` as `readonly` even declaring inside the contrusctor the values

Reply:
It is not possible because I first need the values coming from the filterService which is injected in the constructor.
Instead the readonly keyword can definitely be added 👍🏻